src/python/pants/backend/project_info/cloc.py

```diff
@@ -1,53 +1,49 @@
 # Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
-from typing import cast
+from typing import Tuple
 
 from pants.core.util_rules.external_tool import (
     DownloadedExternalTool,
     ExternalTool,
     ExternalToolRequest,
 )
 from pants.engine.console import Console
-from pants.engine.fs import (
-    CreateDigest,
-    Digest,
-    DigestContents,
-    FileContent,
-    MergeDigests,
-    SourcesSnapshot,
-)
+from pants.engine.fs import Digest, MergeDigests, SourcesSnapshot
 from pants.engine.goal import Goal, GoalSubsystem
 from pants.engine.platform import Platform
 from pants.engine.process import Process, ProcessResult
 from pants.engine.rules import Get, collect_rules, goal_rule
+from pants.option.custom_types import shell_str
+from pants.util.enums import match
 from pants.util.logging import LogLevel
 from pants.util.strutil import pluralize
 
 
 class ClocBinary(ExternalTool):
-    """The cloc lines-of-code counter (https://github.com/AlDanial/cloc)."""
+    """The SCC lines-of-code counter (https://github.com/boyter/scc)."""
 
     options_scope = "cloc-binary"
     name = "cloc"
-    default_version = "1.80"
+    default_version = "2.12.0"
     default_known_versions = [
-        "1.80|darwin|2b23012b1c3c53bd6b9dd43cd6aa75715eed4feb2cb6db56ac3fbbd2dffeac9d|546279",
-        "1.80|linux |2b23012b1c3c53bd6b9dd43cd6aa75715eed4feb2cb6db56ac3fbbd2dffeac9d|546279",
+        "2.12.0|darwin|70b7002cd1e4541cb37b7b9cbc0eeedd13ceacb49628e82ab46332bb2e65a5a6|1842530",
+        "2.12.0|linux|8eca3e98fe8a78d417d3779a51724515ac4459760d3ec256295f80954a0da044|1753059",
     ]
 
-    def generate_url(self, _: Platform) -> str:
+    def generate_url(self, plat: Platform) -> str:
+        plat_str = match(plat, {Platform.darwin: "apple-darwin", Platform.linux: "unknown-linux"})
         return (
-            f"https://github.com/AlDanial/cloc/releases/download/{self.version}/"
-            f"cloc-{self.version}.pl"
+            f"https://github.com/boyter/scc/releases/download/v{self.version}/scc-{self.version}-"
+            f"x86_64-{plat_str}.zip"
         )
 
     def generate_exe(self, _: Platform) -> str:
-        return f"./cloc-{self.version}.pl"
+        return "./scc"
 
 
 class CountLinesOfCodeSubsystem(GoalSubsystem):
-    """Count lines of code."""
+    """Count lines of code using the SCC program (https://github.com/boyter/scc)."""
 
     name = "cloc"
```

Comment:
I'm wondering if the naming here should change. `cloc` indicates a specific tool, that we're no longer using. Possibly this should be `scc`, or something neutral like `countloc`.

Reply:
I think maybe `countloc`? `scc` is pretty vague what it means. What about `count-loc`? We use dashes for some goals like `target-types` and `setup-py`. Makes it look less like a jumble of characters.

Reply:
I'm equally fine with `count-loc` or `countloc`.